xdr/common_test.go

```diff
@@ -0,0 +1,36 @@
+package xdr
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_IDMarshal(t *testing.T) {
+	var id ID
+	rawbytes := bytes.Repeat([]byte{'a'}, 32)
+	copy(id[:], rawbytes)
+	b, err := id.MarshalJSON()
```

Comment:
This should be changed to json.Marshal(id) to help test that the custom marshal function is actually called.